parse.go

```diff
@@ -273,10 +273,12 @@ func (c *Client) Do(req *http.Request, body, result interface{}) (*http.Response
 	}
 
 	res, err := c.RoundTrip(req)
+	if res.Body != nil {
```

Comment:
I'm guessing this was due to the error path in the `RoundTrip` method. The only other possibility is an error from a custom `RoundTripper`. Keeping the current `defer` if we get to line 284 we would `Close` the body as expected.